### ISSUE TITLE
Using with large json inputs is painfully slow

### ISSUE THREAD

**Issue by ircmaxell** [fiatjaf/jiq#11]
The entire application is slowed by large inputs. I am working with a 6mb json file, and each keystroke takes about 500ms to register. I could understand not outputting the results in real-time, but I would suggest separating the UI from the data gathering portions (use a separate UI rendering goroutine)...

**Comment by fiatjaf** [comment]
Yes, that would be a good idea. Unfortunately I don't understand the UI of this thing, I just forked it and changed the json filtering engine to use jq. A PR would be very welcomed.

**Comment by Whoaa512** [comment]
How bout swapping out the exec bit for a pure Go implementation of jq

https://github.com/itchyny/gojq

**Comment by fiatjaf** [comment]
I was thinking of doing that, but I don't think it will improve things by a lot. https://github.com/fiatjaf/jqview uses that and it's painfully slow for big JSON files anyway. Apparently gojq is much slower than raw jq.

I think separating the UI from the JSON processing and also adding some form of debounced typing would improve things more here.

Meanwhile it continues serving as a tool for small and medium JSON analysis only.

**Comment by bayeslearnerold** [comment]
I have a json file about 10M.  
Why is it so slow to move cursor around to change the query?  I haven't done programming for a while, but can you put the computation and the UI in different threads or something?

**Comment by fiatjaf** [comment]
Yes, it is possible, but I don't know how to do it because everything revolves around this TUI library I inherited from jid and don't know how to use.

**Comment by coxley** [comment]
At a minimum, debouncing should resolve a lot of the pain. :)
